### PR TITLE
Chore: Convert QueryOperationRow test to RTL

### LINKED ISSUE #51625: Convert `public/app/core/components/QueryOperationRow/QueryOperationRow.test.tsx` to RTL

### PR DESCRIPTION
**Pull request by Seyaji**
**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/51625

### CODE REVIEW

File: public/app/core/components/QueryOperationRow/QueryOperationRow.test.tsx [hunk 1]
```diff
@@ -1,56 +1,41 @@
...
-        // close
-        titleEl.first().simulate('click');
-      });
+      setup({ isOpen: false, onOpen: onOpenSpy, onClose: onCloseSpy });
+
+      const queryRow = screen.getByText(/^test-title$/);
+      expect(queryRow).toBeInTheDocument();
+
+      fireEvent.click(queryRow);
+      fireEvent.click(queryRow);
```
Comment (Seyaji) [inline public/app/core/components/QueryOperationRow/QueryOperationRow.test.tsx:38]: had trouble getting userEvent.click() to register here so I reverted to using fireEvent instead

**Comment by Seyaji** [comment]
@ashharrison90 Can you make an issue for this? 🙏

File: public/app/core/components/QueryOperationRow/QueryOperationRow.test.tsx [hunk 2]
```diff
@@ -1,56 +1,42 @@
...
-        // close
-        titleEl.first().simulate('click');
-      });
+      setup({ isOpen: false, onOpen: onOpenSpy, onClose: onCloseSpy });
+
+      const queryRow = screen.getByText(/^test-title$/);
+      expect(queryRow).toBeInTheDocument();
+
+      await userEvent.click(queryRow);
+      await userEvent.click(queryRow);
```
Comment (Clarity-89) [inline public/app/core/components/QueryOperationRow/QueryOperationRow.test.tsx:39]: Document that the first click is for opening and second for closing (as done in the original code) would be helpful here.

**Review by Clarity-89 — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
@Seyaji fixed the betterer issue for you. really sorry for the problems with betterer, it _should_ be an invisible check from now on 😄 

this looks great btw! 🥳

**Comment by Seyaji** [comment]
@ashharrison90 no worries! Thanks for sorting it out 🙂

**Pull request merged**